### PR TITLE
[lit-html-server] Fix memory leak when handling result of `unsafeHTML/unsafeSVG` directives

### PR DESCRIPTION
**Pull request by popeindustries**
Repeatedly rendering a template containing an `unsafeHTML/unsafeSVG` directive caused the `templateCache` to grow without limit. The `templateCache` is keyed on the unique and static `TemplateStringsArray` returned when calling a tagged template function. This strings array is guaranteed to be the same across invocations of the function, and therefore is a safe and performant way to cache `Template` instances by relying on referential identity. Unfortunately, it turns out that lit-html returns a _fake_ `TemplateResult` when rendering an `unsafeHTML/unsafeSVG` directive, which breaks identity referencing since each invocation returns a new strings array:

[lit-html/directives.js]
```js
const strings = [value];
// eslint-disable-next-line @typescript-eslint/no-explicit-any
strings.raw = strings;
// WARNING: impersonating a TemplateResult like this is extremely
// dangerous. Third-party directives should not do this.
return (this._templateResult = {
    // Cast to a known set of integers that satisfy ResultType so that we
    // don't have to export ResultType and possibly encourage this pattern.
    // This property needs to remain unminified.
    ['_$litType$']: this.constructor
        .resultType,
    strings,
    values: [],
});
```

 This fix instead returns a `Buffer` of the string content to avoid `Template` processing altogether.

This fake `TemplateResult` technique is also used when handling `innerHTML` string content returned when rendering a custom element, and has been changed to ensure that the same strings array is returned each time.

### CODE REVIEW

**Comment by cristobal** [comment]
Could you release an alpha branch that i can test within our application?

File: packages/lit-html-server/src/internal/parts.js [hunk 1]
```diff
@@ -373,7 +374,11 @@ export class CustomElementOpenPart extends AttributePart {
       if (renderedContent != null) {
         // Handle string from innerHTML (convert to fake TemplateResult to avoid escaping).
         if (typeof renderedContent === 'string') {
-          renderedContent = getFakeTemplateResult(renderedContent);
+          renderedContent = /** @type { TemplateResult } */ ({
+            _$litType$: 1,
+            strings: EMPTY_STRINGS_ARRAY,
+            values: [Buffer.from(renderedContent)],
```
Comment (cristobal) [inline packages/lit-html-server/src/internal/parts.js:380]: Learning along the way here 👍🏽

File: packages/lit-html-server/src/internal/parts.js [hunk 2]
```diff
@@ -613,14 +618,21 @@ async function* resolveAsyncIteratorValue(iterator, tagName, withMetadata) {
...
   const result = directive.render(...directiveResult.values);
 
   if (result === noChange) {
     return EMPTY_STRING_BUFFER;
   }
 
+  if (directiveName === 'unsafeHTML' || directiveName === 'unsafeSVG') {
+    // Return Buffer to avoid string escaping
+    return Buffer.from(result.strings[0]);
+  }
```
Comment (cristobal) [inline packages/lit-html-server/src/internal/parts.js:634]: 📝 👌🏽

**Review by cristobal — APPROVED** [review]

**Comment by cristobal** [comment]
I just tested the compiled code by replacing the existing `lit-html-server.js` code with the one from this branch. 
Into both our application (@janerikbr ) and the experiment, and it works es expected 🔥  

This solves our problem thank you very much @popeindustries 

## Experiment with unsafe
Screenshot 2022-10-08 at 15 06 52

## Our App
Screenshot 2022-10-08 at 15 12 33

It never crashes and the memory never surpases 40MB 👍🏽

**Comment by janerikbr** [comment]
Amazing!

**Pull request merged**